src/auth.ts

```diff
@@ -96,7 +97,12 @@ const generateTokens = async (user: any) => {
     logger.warn("No user found for token generation");
     return {token: null, refreshToken: null};
   }
-  const token = jwt.sign({id: user._id.toString()}, tokenSecretOrKey, getTokenOptions());
+  let payload: Record<string, any> = {id: user._id.toString()};
+  if (authOptions?.generateJWTPayload) {
+    payload = {...payload, ...authOptions.generateJWTPayload(user)};
```

Comment:
So this allows override of the default payload like `id`. We can either just allow that and leave it as is, or prevent it here with reserved keys. I could go either way.

Reply:
Oh that's a good point, I'll flip that.